proposed/2022/version-overrides.md

```diff
@@ -0,0 +1,119 @@
+
+```xml
+<Project>
+  <PropertyGroup>
+    <EnablePackageVersionOverride>false</EnablePackageVersionOverride>
+  </PropertyGroup>
+<Project>
+```
+
+When this is disabled, specifying a `VersionOverride` on a `<PackageReference />` would result in a restore error indicating that the feature is disabled.
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+There could be concerns that allowing users to override package versions for a particular project could lead to version conflicts for transitive project dependencies in the same repository.  While this is true, there are legitimate cases when overriding a centrally defined version for a particular project is the only way to get the desired outcome.  Since users can and will override versions to work around this, we feel its better to give them a supported way of acheiving it.
+
+Although there may be different approaches to the current design, this is one that has been tested in the field with many projects and the feedback has matched our expectations of doing the job to be done in question.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+The current concept that exists in the MSBuildSdks's CentralPackageVersions project has met the needs of many developers around the world who take it on as a dependency and it has been noted many times that this feature is missing in NuGet's first-class support of central package management.
+
+Although there could be a more fleshed out design that includes complete package IDs & their respective versions similar to other ecosystems, we are working with what we know teams need today and will build upon it in the future.   Since the specifications of packages and versions are MSBuild-based, we feel its better to stick with current constructs that Microsoft developer ecosystem customers are familiar with rather than inventing a new process that could lead to confusion.
+
+If for any reason we decide to do a similar "package override" feature where a developer can replace any package on the dependency graph with another, we will allow for different levels of control such as replacing the package ID, the package version, and perhaps other selective mutations for a package in a graph.
+
+Users can also just specify a `<PackageVersion Update="" />` in an import or use an MSBuild property that represents a package version to override what's used in a particular project.
```

Comment:
The `<PackageVersion Update="" />` works fine in the project file itself, as Directory.Packages.props is evaluated first (through the SDK.props chain).  It doesn't need to be in an import.

Reply:
Thanks I keep forgetting that `Directory.Packages.props` is imported before the project.  I'll update this proposal.